vagrant/provisioning/roles/confluent-platform-install/tasks/main.yml

```diff
@@ -1,8 +1,14 @@
 ---
 # tasks file for confluent-install
+- name: Check for current confluent platform version
+  become: yes
+  shell: rpm -qa | grep confluent-platform
+  ignore_errors: yes
+  register: confluent_version
+
 - name: Upgrade confluent platform
   import_tasks: upgrade-confluent.yml
-  when: confluent_major_version is version('6.2', '<')
+  when: confluent_version is undefined or not confluent_version is search("6.2")
```

Comment:
Replace 6.2 with {{ confluent_major_version }}. This way whenever you change the version from what we have, this one will run. Will work for 7.0 for example.